awx/ui_next/src/screens/Job/WorkflowOutput/WorkflowOutput.jsx

```diff
@@ -0,0 +1,222 @@
+import React, { useState, useEffect } from 'react';
+import { withI18n } from '@lingui/react';
+import { t } from '@lingui/macro';
+import styled from 'styled-components';
+import { CardBody as PFCardBody } from '@patternfly/react-core';
+import { layoutGraph } from '@util/workflow';
+import ContentError from '@components/ContentError';
+import ContentLoading from '@components/ContentLoading';
+import { WorkflowJobsAPI } from '@api';
+import WorkflowOutputGraph from './WorkflowOutputGraph';
+
+const CardBody = styled(PFCardBody)`
+  height: calc(100vh - 240px);
+  display: flex;
+  flex-direction: column;
+`;
+
+const Toolbar = styled.div`
+  height: 50px;
+  background-color: grey;
+`;
+
+const Wrapper = styled.div`
+  display: flex;
+  flex-flow: column;
+  height: 100%;
+`;
+
+const fetchWorkflowNodes = async (jobId, pageNo = 1, nodes = []) => {
+  try {
+    const { data } = await WorkflowJobsAPI.readNodes(jobId, {
+      page_size: 200,
+      page: pageNo,
+    });
+    if (data.next) {
+      return await fetchWorkflowNodes(
+        jobId,
+        pageNo + 1,
+        nodes.concat(data.results)
+      );
+    }
+    return nodes.concat(data.results);
+  } catch (error) {
+    throw error;
+  }
+};
+
+function WorkflowOutput({ job, i18n }) {
+  const [contentError, setContentError] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [graphLinks, setGraphLinks] = useState([]);
+  const [graphNodes, setGraphNodes] = useState([]);
+  const [nodePositions, setNodePositions] = useState(null);
+
+  useEffect(() => {
+    const buildGraphArrays = nodes => {
```

Comment:
I think it is worth using an ignore flag for the `react-hooks/exhaustive-deps`  linting rule (if need be) in order to pull some of this logic into an importable and stateless helper w/ unit tests.